demo/src/core/promise-utils.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Creates a debounced version of a function that delays its invocation
+ * until after a specified amount of time has passed since the last time
+ * it was invoked.
+ *
+ * @param {Function} func - The function to be debounced.
+ * @param {number} delay - The time to wait (in milliseconds) before invoking the debounced function.
+ * @returns {Function} - The debounced function.
+ */
+export const debounce = (func, delay) => {
+  let timeoutId;
+
+  return function (...args) {
+    clearTimeout(timeoutId);
+
+    timeoutId = setTimeout(() => {
+      func(...args);
+    }, delay);
+  };
+};
```

Comment:
Debounce the return 2

Reply:
Didn't see that during the rebase, thanks !